### DISCUSSION TITLE
Is there a way to have vd output in the stdout?

### DISCUSSION THREAD

**Discussion by aborruso** [saulpw/visidata#985] [Q&A] [answered]
Hi,
is there a way to run something like the command below (`-of` is only an example parameter)?

```
vd input.csv -b -of json  |  jq '.something'
```

Thank you

**Comment by saulpw** [comment, marked as answer]
Yes; use `vd input.csv -b --save-filetype json | jq ...`

**Reply by aborruso** [reply]
@saulpw  is there a way to hide vd messages?

If I run `vd input.csv -b --save-filetype tsv | grep 'a'` I have

```
opening input.csv as csv
saving 1 sheets to - as tsv
- save finished
id      data
```

Is there a way to have only `id      data`?

Thank you

**Reply by saulpw** [reply]
Those messages go to stderr, so if you redirect stderr then you should only see the data:

    vd [...] 2> /dev/null

**Comment by aborruso** [comment]
Thank you very much: I created a related PR https://github.com/saulpw/visidata.org/pull/46

**Comment by saulbert** [comment]
On the `per-path-filetype` branch (PR #3139), the output format for a conversion can be set explicitly with `-of` / `--output-filetype`, independent of the output name — handy when streaming to stdout, where there is no extension to infer from:

```
vd -b data.csv -of json -o -     # csv → json on stdout
vd -b data.csv -of json | jq .   # same, piped
```

Previously the saver came from the `-o` filename's extension (or the `save_filetype` option); `-of` lets you choose it directly.

[this message written by Claude Opus 4.8 and approved by @saulpw]
